herddb-core/src/main/java/herddb/file/FileDataStorageManager.java

```diff
@@ -326,13 +332,15 @@ public void initTable(String tableSpace, String uuid) throws DataStorageManagerE
                 Bytes value = dataIn.readBytesNoCopy();
                 result.add(new Record(key, value));
             }
-            int pos = dataIn.getPosition();
             long hashFromFile = dataIn.readLong();
-            // after the hash we will have zeroes or garbage
-            // the hash is not at the end of file, but after data
-            long hashFromDigest = XXHash64Utils.hash(dataPage, 0, pos);
-            if (hashFromDigest != hashFromFile) {
-                throw new DataStorageManagerException("Corrupted datafile " + pageFile + ". Bad hash " + hashFromFile + " <> " + hashFromDigest);
+            if (HASH_CHECKS_ENABLED && hashFromFile > 0) {
```

Comment:
"hashFromFile != 0"

please remember that the hash can be a negative number

it would be better to introduce a constant like "NO_HASH_PRESENT" = 0
this way the code will look better and more self explaining